### PR TITLE
Core 5.0.0 support

### PR DESCRIPTION
**Pull request by ekkinox**
* Updated oat-sa/lib-lti1p3-core dependency to version 5.0
* Updated documentation

target release: `5.0.0`

### CODE REVIEW

**Review by GregoryGagua — APPROVED** [review]

File: Security/Authentication/Token/Service/LtiServiceSecurityToken.php [hunk 1]
```diff
@@ -24,25 +24,25 @@
...
 {
     /** @var string[] */
     private $roleNames;
 
-    /** @var AccessTokenRequestValidationResult|null */
+    /** @var RequestAccessTokenValidationResult|null */
     private $validationResult;
 
-    public function __construct(AccessTokenRequestValidationResult $validationResult = null)
+    public function __construct(RequestAccessTokenValidationResult $validationResult = null)
```
Comment (siwane) [inline Security/Authentication/Token/Service/LtiServiceSecurityToken.php]: optional args, please add leading `?`

File: Security/Authentication/Token/Service/LtiServiceSecurityToken.php [hunk 2]
```diff
@@ -80,7 +80,7 @@ public function getRoleNames(): array
         return $this->roleNames;
     }
 
-    private function applyValidationResult(AccessTokenRequestValidationResult $validationResult = null): void
+    private function applyValidationResult(RequestAccessTokenValidationResult $validationResult = null): void
```
Comment (siwane) [inline Security/Authentication/Token/Service/LtiServiceSecurityToken.php]: optional args, please add leading `?`

File: composer.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "require": {
         "php": ">=7.2.0",
         "ext-json": "*",
-        "oat-sa/lib-lti1p3-core": "^4.0",
+        "oat-sa/lib-lti1p3-core": "dev-feature/service-handlers",
```
Comment (siwane) [inline composer.json]: not ready for merge

File: Security/Authentication/Token/Service/LtiServiceSecurityToken.php (hunk 1 above)
Comment (ekkinox) [reply Security/Authentication/Token/Service/LtiServiceSecurityToken.php]: done

File: Security/Authentication/Token/Service/LtiServiceSecurityToken.php (hunk 2 above)
Comment (ekkinox) [reply Security/Authentication/Token/Service/LtiServiceSecurityToken.php]: done

File: composer.json (hunk 3 above)
Comment (ekkinox) [reply composer.json]: now based on core 5.0.0, we can merge once approved

**Review by siwane — APPROVED** [review]

**Pull request merged**